diff --git a/lib/comgr/test/CMakeLists.txt b/lib/comgr/test/CMakeLists.txt
index 9fe723f..e358d28 100644
--- a/lib/comgr/test/CMakeLists.txt
+++ b/lib/comgr/test/CMakeLists.txt
@@ -101,6 +101,10 @@ if (DEFINED HIP_COMPILER AND "${HIP_COMPILER}" STREQUAL "clang")
   target_compile_definitions("${name}"
     PRIVATE -DHIP_COMPILER=${HIP_COMPILER})
 endif()
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    target_compile_definitions("${name}"
+      PRIVATE -D_CRT_SECURE_NO_WARNINGS)
+  endif()
   target_link_libraries("${name}"
     amd_comgr)
   add_dependencies("${name}"
